File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        ret = []
        nums.sort()
        for i in range(len(nums)):
            if i > 0 and nums[i] == nums[i - 1]:
                continue
            target = -nums[i]
            L = i + 1
            R = len(nums) - 1
            while L < R:
                if nums[L] + nums[R] > target:
                    R = R - 1
                elif nums[L] + nums[R] < target:
                    L = L + 1
                else:
                    l = [nums[i], nums[L], nums[R]]
                    ret.append(l)
                    L += 1
                    R -= 1
                    while L < R and nums[L] == nums[L - 1]:
                            L += 1
                    while L < R and nums[R] == nums[R + 1]:
                        R -= 1
        return ret


            

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
public class Solution 
{
    public int[] topKFrequent(int[] nums, int k) 
    {
        Map<Integer, Integer> count = new HashMap<>();
        for (int num : nums) 
        {
            count.put(num, count.getOrDefault(num, 0) + 1);
            //puts the number and its count into the hashmap (key, value)
            //count.getOrDefault(num, 0) + 1) if num(key) exists in 
            //the map already we add a number to that value, if not
            //we return 0 and add 1 to it (updates the count for the number)
        }

        List<int[]> arr = new ArrayList<>(); //makes arraylist of integer arrays
        for (Map.Entry<Integer, Integer> entry : count.entrySet()) 
        {
            arr.add(new int[] {entry.getValue(), entry.getKey()});
            //adds a new int array for every hashmap entry [value, num], [count, number]
        }
        arr.sort((a, b) -> b[0] - a[0]);
        //looks at the frequency of 2 elements in the arraylist
        //b[0] - a[0] sorts by greater numbers, most freq. first]

        int[] res = new int[k];
        //makes array to get the k most freq numbers
        for (int i = 0; i < k; i++) 
        {
            res[i] = arr.get(i)[1];
            //puts the number only into the array, not its count
        }
        return res;
    }
}
